test/functional/tests/stats/test_statistics_integrity.py

```diff
@@ -0,0 +1,704 @@
+                        f"for '{stat_name}' is {stat_value}, should equal 0\n")
+
+
+def check_stats_after_io(caches, cores, after_reload: bool = False):
+    for i in range(caches_count):
+        cache_mode = caches[i].get_cache_mode()
+        cores_stats = [
+            cores[i][j].get_statistics(stat_filter=stat_filter)
+            for j in range(cores_per_cache)
+        ]
+        cores_stats_perc = [
+            cores[i][j].get_statistics(stat_filter=stat_filter, percentage_val=True)
+            for j in range(cores_per_cache)
+        ]
+        cores_error_stats, cores_error_stats_perc = get_stats_flat(
+            cores[i], stat_filter=[StatsFilter.err]
+        )
+        for j in range(cores_per_cache):
+            fail_message = (
+                f"For core device {cores[i][j].system_path} in {cache_mode} cache mode ")
+            if after_reload:
+                validate_usage_stats(
+                    cores_stats[j], cores_stats_perc[j], caches[i], cache_mode, fail_message)
+                validate_error_stats(
+                    cores_error_stats[j], cores_error_stats_perc[j], cache_mode, fail_message)
+            else:
+                validate_usage_stats(
+                    cores_stats[j], cores_stats_perc[j], caches[i], cache_mode, fail_message)
+                validate_request_stats(
+                    cores_stats[j], cores_stats_perc[j], cache_mode, fail_message)
+                validate_block_stats(
+                    cores_stats[j], cores_stats_perc[j], cache_mode, fail_message)
+                validate_error_stats(
+                    cores_error_stats[j], cores_error_stats_perc[j], cache_mode, fail_message)
+
+
+def check_stats_sum(caches, cores):
+    for i in range(caches_count):
+        cores_stats, cores_stats_perc, cache_stats = (
+            get_stats_flat(cores[i], cache=caches[i])
+        )
+        for cache_stat_name in cache_stats.keys():
+            if cache_stat_name.lower() != "free":
+                core_stat_name = cache_stat_name.replace("(s)", "")
+                core_stat_sum = 0
+                try:
+                    cache_stats[cache_stat_name] = cache_stats[cache_stat_name].value
+                    for j in range(cores_per_cache):
+                        cores_stats[j][core_stat_name] = cores_stats[j][core_stat_name].value
+                except AttributeError:
+                    pass
+                for j in range(cores_per_cache):
+                    core_stat_sum += cores_stats[j][core_stat_name]
+                if core_stat_sum != cache_stats[cache_stat_name]:
+                    TestRun.LOGGER.error(
+                        f"For cache ID {caches[i].cache_id} sum of core's "
+                        f"'{core_stat_name}' values is {core_stat_sum}, "
+                        f"should equal {cache_stats[cache_stat_name]}\n")
+
+
+def validate_usage_stats(stats, stats_perc, cache, cache_mode, fail_message):
```

Comment:
Just a suggestion, I will not enforce this ;) I'm wondering about the stat checks. I think it would be beneficial, to group checks by stats and not by traits. This would show what the expected value is for a given stat depending on traits. But will probably be more complicated in code than current way. I started from a question - what would be the 'clean' value for different modes, and 'clean' is checked in 3 different places.